File: src/domain/model/user/aggregate.ts
import { UserId } from './userId';

export class UserAggregate {
  private _id: UserId;

  constructor(_id: UserId = new UserId()) {
    this._id = _id;
  }

  getIdOfPrivateValue(): UserId {
    return this._id;
  }

  static userIdFromStr(id: string): UserId {
    return new UserId(id);
  }
}

File: src/domain/model/user/aggregate.rs
use super::user_aggregate_id::UserAggregateId;

#[derive(Debug, Clone, Eq, Hash, PartialOrd, PartialEq)]
pub struct UserAggregate {
  id: UserAggregateId,
}

impl UserAggregate {
  pub fn new(user_id: &UserAggregateId) -> Self {
    UserAggregate { id: user_id.clone() }
  }
}
